src/main/java/com/teragrep/rlp_11/RelpProbeConfiguration.java

```diff
@@ -0,0 +1,169 @@
+            throw new RelpProbeConfigurationError(
+                    "Invalid <event.delay> property, expected > 0, received <[" + eventDelay + "]>"
+            );
+        }
+
+        getAndVerifyStringProperty("target.hostname");
+
+        int targetPort = getAndVerifyNumberProperty("target.port");
+        if (targetPort <= 0 || targetPort > MAXIMUM_PORT) {
+            LOGGER
+                    .error(
+                            "Invalid <target.port> property, expected between 1 and {}, received <[{}]>", MAXIMUM_PORT,
+                            targetPort
+                    );
+            throw new RelpProbeConfigurationError(
+                    "Invalid <target.port> property, expected between 1 and " + MAXIMUM_PORT + ", received <["
+                            + targetPort + "]>"
+            );
+        }
+
+        int prometheusPort = getAndVerifyNumberProperty("prometheus.port");
+        if (prometheusPort <= 0 || prometheusPort > MAXIMUM_PORT) {
+            LOGGER
+                    .error(
+                            "Invalid <prometheus.port> property, expected between 1 and {}, received <[{}]>",
+                            MAXIMUM_PORT, prometheusPort
+                    );
+            throw new RelpProbeConfigurationError(
+                    "Invalid <prometheus.port> property, expected between 1 and " + MAXIMUM_PORT + ", received <["
+                            + prometheusPort + "]>"
+            );
+        }
+
+        int reconnectInterval = getAndVerifyNumberProperty("target.reconnectinterval");
+        if (reconnectInterval <= 0) {
+            LOGGER
+                    .error(
+                            "Invalid <target.reconnectinterval> property, property, expected > 0, received <[{}]>",
+                            reconnectInterval
+                    );
+            throw new RelpProbeConfigurationError(
+                    "Invalid <target.reconnectinterval> property, expected > 0, received <[" + reconnectInterval + "]>"
+            );
+        }
+    }
+
+    private String getAndVerifyStringProperty(String name) {
+        String property = config.get(name);
+        if (property == null) {
+            LOGGER.error("Missing <{}> property", name);
+            throw new RelpProbeConfigurationError("Missing <" + name + "> property");
+        }
+        return property;
+    }
+
+    private int getAndVerifyNumberProperty(String name) {
+        String property = getAndVerifyStringProperty(name);
+        try {
+            return Integer.parseInt(property);
+        }
+        catch (NumberFormatException e) {
+            LOGGER.error("Invalid <{}> property received, not a number: <{}>", name, e.getMessage());
+            throw new RelpProbeConfigurationError(
+                    "Invalid <" + name + "> property received, not a number: <" + e.getMessage() + ">"
+            );
+        }
+    }
+
+    public String getEventHostname() {
```

Comment:
remove get, create separate EventConfig, use separate EventFactory if necessary in the usage. see the separate event construction into different method comment.